src/utils/formatNumbers.ts

```diff
@@ -4,16 +4,32 @@ import { DEFAULT_LOCALE } from 'constants/locales'
 import numbro from 'numbro'
 
 // Convert [CurrencyAmount] to number with necessary precision for price formatting.
-export const currencyAmountToPreciseFloat = (currencyAmount: CurrencyAmount<Currency>) => {
-  const floatForLargerNumbers = parseFloat(currencyAmount.toFixed(3))
+export const currencyAmountToPreciseFloat = (currencyAmount: CurrencyAmount<Currency> | undefined) => {
+  if (!currencyAmount) return undefined
+  const floatForLargerNumbers = parseFloat(currencyAmount.toExact())
   if (floatForLargerNumbers < 0.1) {
-    return parseFloat(currencyAmount.toSignificant(3))
+    return parseFloat(currencyAmount.toSignificant(6))
   }
-  return floatForLargerNumbers
+  return parseFloat(currencyAmount.toExact())
+}
+
+interface FormatDollarArgs {
+  num: number | undefined | null
+  isPrice?: boolean
+  neater?: boolean
+  digits?: number
+  round?: boolean
 }
 
 // Using a currency library here in case we want to add more in future.
-export const formatDollar = (num: number | undefined | null, isPrice = false, digits = 2, round = true) => {
+export const formatDollar = ({
+  num,
+  isPrice = false,
+  neater = false,
```

Comment:
truncateEarly? would prefer a more descriptive term

Reply:
used lessPreciseStablecoinValues and added a comment to explain